spec/controllers/hq/sponsors_controller_spec.rb

```diff
@@ -1,38 +1,47 @@
 require 'rails_helper'
+require 'will_paginate/array'
 
 RSpec.describe Hq::SponsorsController, type: :controller do
+    let(:sponsor) {FactoryGirl.build_stubbed :sponsor}
+    let(:sponsors) {(1..5).map {FactoryGirl.build_stubbed :sponsor}}
+
   before :each do
     sign_in instance_double(AdminUser)
-    @sponsor = FactoryGirl.build_stubbed :sponsor
-    @sponsor_build = FactoryGirl.build :sponsor
-    @sponsor_new = Sponsor.new
   end
 
   specify '#index' do
-    expect(Sponsor).to receive(:all).and_return( [ @sponsor ] )
-    get :index
+    expect(Sponsor).to receive(:paginate).with(page: "2").
+                  and_return( sponsors.paginate(per_page: 2) )
+    get :index, page: 2
+    expect(assigns(:sponsors).count).to eq 2
     expect(response).to render_template 'index'
   end
 
   specify '#show' do
-    expect(Sponsor).to receive(:find).and_return(@sponsor)
-    get :show, id: @sponsor.id
-    expect(assigns(:sponsor)).to eq @sponsor
+    expect(Sponsor).to receive(:find).and_return(sponsor)
+    get :show, id: sponsor.id
+    expect(assigns(:sponsor)).to eq sponsor
     expect(response).to render_template 'show'
   end
 
   specify '#new' do
-    expect(Sponsor).to receive(:new).and_return(@sponsor_new)
+    sponsor_new = Sponsor.new
+    expect(Sponsor).to receive(:new).and_return(sponsor_new)
     get :new
     expect(assigns(:sponsor)).to be_a_new(Sponsor)
     expect(response).to render_template 'new'
   end
 
   context '#create' do
+    before :each do
+      @sponsor_new = Sponsor.new
+      @sponsor_build = FactoryGirl.build :sponsor
+    end
+
     specify 'successful' do
       expect(Sponsor).to receive(:new).and_return(@sponsor_new)
       expect {post :create, sponsor: @sponsor_build.attributes }.to change{Sponsor.count}.by 1
-      expect(response).to redirect_to hq_sponsor_url(Sponsor.find_by_name(@sponsor_build.name))
+      expect(response).to redirect_to hq_sponsor_url( Sponsor.find_by_name(@sponsor_build.name) )
```

Comment:
I realize this is not a part of this PR, but the test on L43 does not seem to belong in a controller spec. It exercises not only the controller but the model and the db as well - something that should be reserved for integration-level testing. What are people's thoughts?